apps/wallet/src/shared/deepBook/context.tsx

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { useActiveAccount } from '_app/hooks/useActiveAccount';
+import { useGetOwnedObjects } from '@mysten/core';
+import { useSuiClient } from '@mysten/dapp-kit';
+import { DeepBookClient } from '@mysten/deepbook';
+import { createContext, useContext, useMemo, type ReactNode } from 'react';
+
+type DeepBookContextProps = {
+	client: DeepBookClient;
+	accountCapId: string;
+};
+
+const DeepBookContext = createContext<DeepBookContextProps | null>(null);
+
+interface DeepBookContextProviderProps {
+	children: ReactNode;
+}
+
+export function useDeepBookContext() {
+	const context = useContext(DeepBookContext);
+	if (!context) {
+		throw new Error('useDeepBookContext must be used within a DeepBookContextProvider');
+	}
+	return context;
+}
+
+export function useDeepBookClient() {
```

Comment:
I have a suggestion to remove some of this code anyway, but if this survives, you don't make these granular hooks, just keep the `useDeepbook` hook.